bwc-test/build.gradle

```diff
@@ -76,13 +76,13 @@ dependencies {
 String bwcVersion = "1.3.0.0";
 String baseName = "securityBwcCluster"
 String bwcFilePath = "src/test/resources/"
-String projectVersion = "2.0.0.0"
+String projectVersion = "2.1.0.0"
 
 2.times {i ->
     testClusters {
         "${baseName}$i" {
             testDistribution = "ARCHIVE"
-        versions = ["1.3.0","2.0.0"]
```

Comment:
Should these versions be additive instead of replacing the previous version?

Reply:
Upgrading to `2.0.0` is tested on `2.0` branch.

Reply:
My concern is if we add an incompatibility in version in 2.1 that doesn't work with 2.0.  Can you help me find where we test this scenario?

Reply:
That's a great point! The test should validate the upgrade from 2.0.0 to 2.1.0.

Reply:
According to our support matrix, we support the last major and the current previous major, so we should have tests for `1.3 -> 2.1` and `2.0 -> 2.1`, can we make sure we test both?

stated differently, we should have tests that cover: 
* last supported major version (1) and its most recent minor version (3) to current version
* current major version (2) and its most recent minor version (0) to current version 

if we really wanted to be thorough I think we would add checks for every minor version on the current major version (e.g. if we were on 2.4, we would have BWC tests for 1.3, 2.0, 2.1, 2.2, and 2.3) (e.g.2 if we were on 3.1 we would have BWC tests for 2.3 and 3.0)

I think we should codify your preferred plan for this is in a comment side by side with these values to make future changes clear.

Reply:
>the current previous major

Do you mean current previous minor?

> According to our support matrix, we support the last major and the current previous major, so we should have tests for 1.3 -> 2.1 and 2.0 -> 2.1

Could you post the link to the support matrix? I don't see any plugins testing upgrade from multiple previous versions to the current version though.

Reply:
the current previous major

Do you mean current previous minor?

According to our support matrix, we support the last major and the current previous major, so we should have tests for 1.3 -> 2.1 and 2.0 -> 2.1

Could you post the link to the support matrix? We should follow if there is a consistent requirements, but I don't see any plugins testing upgrade from multiple previous versions to the current version though.

Reply:
@cliu123 I've created https://github.com/opensearch-project/OpenSearch/issues/3440 to follow up for broad guidance.  Lets move ahead with the work you have done here.